src/main/java/core/basesyntax/strategy/operation/PurchaseOperation.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax.strategy.operation;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void handle(FruitTransaction fruit) {
+        String getFruitFromStorage = fruit.getName();
+        Integer quantityBefore = Storage.fruitStorage.get(getFruitFromStorage);
+        if (quantityBefore == null) {
+            throw new RuntimeException(fruit.getName()
+                + "completed.");
+        }
+        if (quantityBefore < fruit.getQuantity()) {
+            throw new RuntimeException("We have run out of"
+                + fruit.getName()
+                            + ". Only"
+                + quantityBefore
+                            + "amount is left.");
+        }
```

Comment:
You can make some private method like validation

Reply:
I have create method 'validateTransaction' with a private success modifier and a void keyword. This method takes two parameter : an instance of the 'FruitTransaction' model class called 'fruit' and the quantity of items in the storage before transaction begins called 'quantityBefore'. In the method, we perform check to see if the quantity of items before begging transaction is equals zero and if the quantity of items before transaction begins is less than the specified quantity. If it is the truth, we throw a RuntimeException with an error explanations. After that we call the 'validateTransaction' from the overrided 'handle' method.

Reply:
`
public class PurchaseOperation implements OperationHandler {
    @Override
    public void handle(FruitTransaction fruit) {
        String getFruitFromStorage = fruit.getName();
        Integer quantityBefore = Storage.fruitStorage.get(getFruitFromStorage);

        validateTransaction(fruit, quantityBefore);

        Integer quantityAfter = quantityBefore - fruit.getQuantity();
        Storage.fruitStorage.put(getFruitFromStorage, quantityAfter);
    }

    private void validateTransaction(FruitTransaction fruit, Integer quantityBefore) {
        if (quantityBefore == null) {
            throw new RuntimeException(fruit.getName()
                    + "completed.");
        }
        if (quantityBefore < fruit.getQuantity()) {
            throw new RuntimeException("We have run out of"
                    + fruit.getName() + ". Only"
                    + quantityBefore + "amount is left.");
        }
    }
}`